src/main/resources/public/assets/scripts/cookieControl.js

```diff
@@ -0,0 +1,136 @@
+(function() {
+  var elsCookieControl = {
+    prefsCookieName: 'els-cookies-allowed',
+    nonEssentialCookieNames: [
+      '_gat_govuk_shared',
+      '_ga',
+      '_gat',
+      '_gid'
+    ],
+    cookieScripts: [
+      // Google Analytics
+      function() {
+        if(window.ELS.googleAnalyticsEnabled) {
+          (function (i, s, o, g, r, a, m) {
+            i['GoogleAnalyticsObject'] = r;
+            i[r] = i[r] || function () {
+              (i[r].q = i[r].q || []).push(arguments)
+            }, i[r].l = 1 * new Date();
+            a = s.createElement(o),
+              m = s.getElementsByTagName(o)[0];
+            a.async = 1;
+            a.src = g;
+            m.parentNode.insertBefore(a, m)
+          })(window, document, 'script', 'https://www.google-analytics.com/analytics.js', 'ga');
+
+          ga('create', 'UA-145652997-1', 'auto', 'govuk_shared', {'allowLinker': true});
+          ga('govuk_shared.require', 'linker');
+          ga('govuk_shared.linker.set', 'anonymizeIp', true);
+          ga('govuk_shared.linker:autoLink', ['www.gov.uk']);
+
+          ga('create', 'UA-136887405-1', 'auto');
+          ga('send', 'pageview');
+          ga('govuk_shared.send', 'pageview');
+
+          if(document.getElementById('googleAnalyticsClientId')) {
+            ga(function (tracker) {
+              var clientId = tracker.get('clientId');
+              document.getElementById('googleAnalyticsClientId').value = clientId;
+            });
+          }
+        } else {
+          console.warn("You've consented to analytics cookies but Google Analytics integration is disabled on this environment, so the Google Analytics script won't be loaded.");
+        }
+      }
+    ],
+    setCookie: function (name,value,days) {
+      var expires = "";
+      if (days) {
+        var date = new Date();
+        date.setTime(date.getTime() + (days*24*60*60*1000));
+        expires = "; expires=" + date.toUTCString();
+      }
+      document.cookie = name + "=" + (value || "")  + expires + "; path=/";
+    },
+    getCookie: function (name) {
+      var nameEQ = name + "=";
+      var ca = document.cookie.split(';');
+      for(var i=0;i < ca.length;i++) {
+        var c = ca[i];
+        while (c.charAt(0)==' ') c = c.substring(1,c.length);
+        if (c.indexOf(nameEQ) == 0) return c.substring(nameEQ.length,c.length);
+      }
+      return null;
+    },
+    eraseCookie: function (name) {
+      document.cookie = name+'=; Max-Age=-99999999;';
```

Comment:
Should this include the path? Appreciate it's very edge case, but you may have the same cookie name under different paths for the same domain which are both in scope and you want to delete a specific one (e.g. `domain` and `domain/some-other-app`). Perhaps `nonEssentialCookieNames` should be an array of name+path objects if we want this to be a reusable generic solution elsewhere?

Reply:
Good point, probably won't happen here but I'll fix in case this gets ported into FDS